common/pinot/pinotQueryValidator_test.go

```diff
@@ -113,6 +113,43 @@ func TestValidateQuery(t *testing.T) {
 			query:     "CustomIntField = 1 or CustomIntField = 2",
 			validated: "(JSON_MATCH(Attr, '\"$.CustomIntField\"=''1''') or JSON_MATCH(Attr, '\"$.CustomIntField\"=''2'''))",
 		},
+		"Case14-1: range query: custom filed": {
+			query:     "CustomIntField BETWEEN 1 AND 2",
+			validated: "(JSON_MATCH(Attr, '\"$.CustomIntField\" is not null') AND CAST(JSON_EXTRACT_SCALAR(Attr, '$.CustomIntField') AS INT) >= 1 AND CAST(JSON_EXTRACT_SCALAR(Attr, '$.CustomIntField') AS INT) <= 2)",
+		},
+		"Case14-2: range query: system filed": {
+			query:     "NumClusters BETWEEN 1 AND 2",
+			validated: "NumClusters between 1 and 2",
+		},
+		"Case15-1: custom date attribute less than": {
+			query:     "CustomDatetimeField < 1697754674",
+			validated: "(JSON_MATCH(Attr, '\"$.CustomDatetimeField\" is not null') AND CAST(JSON_EXTRACT_SCALAR(Attr, '$.CustomDatetimeField') AS BIGINT) < 1697754674)",
+		},
+		"Case15-2: custom date attribute greater than or equal to": {
+			query:     "CustomDatetimeField >= 1697754674",
+			validated: "(JSON_MATCH(Attr, '\"$.CustomDatetimeField\" is not null') AND CAST(JSON_EXTRACT_SCALAR(Attr, '$.CustomDatetimeField') AS BIGINT) >= 1697754674)",
+		},
+		"Case15-3: system date attribute greater than or equal to": {
+			query:     "StartTime >= 1697754674",
```

Comment:
When user input a date string like StartTime > "2023-11-02T15:36:47", is this handled in the query validator?

Reply:
This is not handled here. But that will be converted to `UnixMilli()` in PiontVisibilityStore.